### PR TITLE
Burial 90229 Refactor away from CentralMail

### PR DESCRIPTION
**Pull request by wayne-weibel**
## Summary

remove reference to CentralMail in Burials code

## Related issue(s)

[Burial | Refactor away from CentralMail](https://github.com/department-of-veterans-affairs/va.gov-team/issues/90229)
https://github.com/department-of-veterans-affairs/vets-website/pull/33019

## Testing done

- [x] *New code is covered by unit tests*
- local submission of a Burial form via website and verify entry to the database

## What areas of the site does it impact?

Burial

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected

### CODE REVIEW

**Review by balexandr — DISMISSED** [review]
LGTM

File: app/controllers/v0/burial_claims_controller.rb [hunk 1]
```diff
@@ -4,26 +4,31 @@
 require 'burials/monitor'
```
Comment (RachalCassity) [inline app/controllers/v0/burial_claims_controller.rb:5]: Is the BurialClaimsController missing`require 'common/exceptions/validation_errors'`? ClaimsBaseController had `require 'common/exceptions/validation_errors'` and ApplicationController does not

File: app/controllers/v0/burial_claims_controller.rb [hunk 2]
```diff
@@ -4,26 +4,31 @@
...
       monitor.track_show404(params[:id], current_user, e)
-      render(json: { data: { attributes: { state: 'not found' } } }, status: :not_found)
+      render(json: { error: e.to_s }, status: :not_found)
     rescue => e
       monitor.track_show_error(params[:id], current_user, e)
-      render(json: { data: { attributes: { state: 'error processing request' } } }, status: :unprocessable_entity)
+      raise e
     end
 
     def create
```
Comment (RachalCassity) [inline app/controllers/v0/burial_claims_controller.rb:34]: I do not see a create_claim route but the controller is calling create_claim on line 32. Please move create_claim to private.

**Review by RachalCassity — CHANGES_REQUESTED** [review]

File: spec/controllers/v0/burial_claims_controller_spec.rb [hunk 3]
```diff
@@ -38,48 +34,43 @@ def send_create
...
+      response = get(:show, params: { id: 'non-existant-saved-claim' })
 
-      expect(response).to have_http_status(:unprocessable_entity)
+      expect(response.status).to eq(500)
     end
   end
 
   describe '#process_and_upload_to_lighthouse' do
-    let(:claim) { build(:pensions_module_pension_claim) }
+    let(:claim) { build(:burial_claim_v2) }
```
Comment (RachalCassity) [inline spec/controllers/v0/burial_claims_controller_spec.rb:73]: I do not see the burial_claim factory being used anywhere. Should burial_claim_v2 replace burial_claim?  Looks like formV2 just needs to be added to the burial_claim factory

**Review by RachalCassity — CHANGES_REQUESTED** [review]

File: spec/controllers/v0/burial_claims_controller_spec.rb (hunk 3 above)
Comment (wayne-weibel) [reply spec/controllers/v0/burial_claims_controller_spec.rb:73]: Not sure what you mean by this - `burial_claim_v2` is what should be used here, and is present in the factory `spec/factories/burial_claim.rb` line 36

Comment (RachalCassity) [reply spec/controllers/v0/burial_claims_controller_spec.rb:73]: My bad. I see others using the original

**Review by RachalCassity — DISMISSED** [review]

**Review by rjohnson2011 — APPROVED** [review]

**Pull request merged**